src/matching.rs

```diff
@@ -284,6 +284,20 @@ impl<T> DeclarationBlock<T> {
     }
 }
 
+bitflags! {
+    #[doc = "Flags set on elements during the matching process."]
+    flags ElementFlags: u8 {
+        #[doc = "When a child is added or removed from this element, all the children must be"]
+        #[doc = "restyled, because they may match selectors such as :last-of-type."]
+        const HAS_SLOW_SELECTOR = 0x01,
+        #[doc = "When a child is added or removed from this element, any later children must be"]
+        #[doc = "restyled, because they may match selectors such as :nth-child."]
+        const HAS_SLOW_SELECTOR_LATER_SIBLINGS = 0x02,
```

Comment:
Please add an exhaustive list of which pseudo-classes set which flags. Is it all child-indexed and type-child-indexed pseudos? (See table of contents of https://drafts.csswg.org/selectors/) If there’s a flag for later siblings, why not one for earlier siblings with `:nth-last-child()`?